lib/openshift/flakes/related_object.rb

```diff
@@ -0,0 +1,27 @@
+module BushSlicer
+
+  # this class should help with parsing status.relatedObjects
+  # https://github.com/openshift-qe/output_references/blob/master/clusteroperators/service-ca-operator.yaml
+  class RelatedObject
+
+    attr_reader :struct
+    private :struct
+
+    def initialize(struct)
```

Comment:
I would suggest to check `lib/openshift/flakes/image_ref.rb`. This class can convert the object to actual resource. Or fail if such resource class is not defined already.